crowbar_framework/lib/crowbar/error/upgrade.rb

```diff
@@ -0,0 +1,54 @@
+#
+# Copyright 2017, SUSE LINUX GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+module Crowbar
+  module Error
+    # generic upgrade
+    class UpgradeError < StandardError
+    end
+
+    # cancel upgrade
+    class UpgradeCancelError < StandardError
+      def initialize(step_name = "")
+        super("Not possible to cancel the upgrade at the step #{step_name}")
+      end
+    end
+
+    # openstack backup
+    class UpgradeNotEnoughDiskSpaceError < StandardError
+      def initialize
+        super("Not enough space in /var/lib/crowbar/upgrade to create a OpenStack database dump.")
+      end
+    end
+
+    class UpgradeFreeDiskSpaceError < StandardError
+      def initialize(message = "")
+        super("Cannot determine free disk space. #{message}")
```

Comment:
This is shown as
``steps.backup_openstack.errors.backup_openstack.data | Cannot determine free disk space. 8544944128``

Please wrap the size into some text.